gcovr/utils.py

```diff
@@ -315,14 +315,34 @@ def percent_uncovered_key(key):
 
 
 @contextmanager
-def open_binary_for_writing(filename=None):
+def open_text_for_writing(filename=None, **kwargs):
```

Comment:
I saw that binary function was already present and each format use its own code for deciding where to print (STDOUT or file). Also the binary function prints to STDOUT if the filename `-` is given. Thats the reason for introducing the text function and use this functions for all formats.

I also thought about this way, but e.g. the JSON format has two print_... functions which use different names and call the function `_write_json_result` with the full path. Then we have to add a argument for the default name. I'll implement this because this is also removing duplicated code.